middleware/recover.go

```diff
@@ -113,7 +118,11 @@ func RecoverWithConfig(config RecoverConfig) echo.MiddlewareFunc {
 							c.Logger().Print(msg)
 						}
 					}
-					c.Error(err)
+
+					if(!config.DisableErrorHandler) {
+						c.Error(err)
+					}
+					returnErr = err
```

Comment:
I think calling error( and setting that return error should be mutally exclusive - this is how it works today (error is "swallowed"

```go
					if(!config.DisableErrorHandler) {
						c.Error(err)
					} else {
					returnErr = err
}
```

and that first test can be changed to check that no error is returned
```go
func TestRecover(t *testing.T) {
	e := echo.New()
	buf := new(bytes.Buffer)
	e.Logger.SetOutput(buf)
	req := httptest.NewRequest(http.MethodGet, "/", nil)
	rec := httptest.NewRecorder()
	c := e.NewContext(req, rec)
	h := Recover()(func(c echo.Context) error {
		panic("test")
	})
	err := h(c)
	assert.NoError(t, err)
	assert.Equal(t, http.StatusInternalServerError, rec.Code)
	assert.Contains(t, buf.String(), "PANIC RECOVER")
}
```

Reply:
Yes, error should be thrown back if it's not handled by recovery middleware and otherwise swallowed. I have updated it correctly now.